lib/cpp/CMakeLists.txt

```diff
@@ -23,6 +23,10 @@ REQUIRE_BOOST_HEADERS()
 
 include_directories(src)
 
+if(NOT BUILD_SHARED_LIBS)
+  add_definitions("-DTHRIFT_STATIC_DEFINE")
```

Comment:
Seems oddly named... suggestion...
```
if (BUILD_SHARED_LIBS)
    add_definitions("-DTHRIFT_LINK_SHARED")
endif ()

Reply:
THRIFT_STATIC_DEFINE is generated by cmake.In order to switch to cmake smoothly in the future,I don't want to change it.